tests/dotnet/UnitTests/ProjectTest.cs

```diff
@@ -1647,6 +1647,30 @@ public void PublishAotDuringBuild (ApplePlatform platform, string runtimeIdentif
 			DotNet.AssertBuild (project_path, properties);
 		}
 
+		[Test]
+		[TestCase (ApplePlatform.iOS, "ios-arm64")]
+		[TestCase (ApplePlatform.iOS, "iossimulator-arm64")]
+		[TestCase (ApplePlatform.MacCatalyst, "maccatalyst-arm64")]
+		public void PublishAotWithWarningInDebugBuild (ApplePlatform platform, string runtimeIdentifiers)
```

Comment:
nit: I would remove the `Debug` in the name as we expect to get the same behaviour in release mode as well i.e. doing:
`dotnet build -c Release`
`dotnet build -c Debug`
should have the same effect -> see trim warnings if `PublishAot=true`